pkg/platform/kube/platform.go

```diff
@@ -639,21 +639,37 @@ func (p *Platform) RedeployFunction(ctx context.Context, redeployFunctionOptions
 	return nil
 }
 
-func (p *Platform) GetFunctionReplicaLogsStream(ctx context.Context,
-	options *platform.GetFunctionReplicaLogsStreamOptions) (io.ReadCloser, error) {
-	return p.consumer.KubeClientSet.
-		CoreV1().
-		Pods(options.Namespace).
-		GetLogs(options.Name, &v1.PodLogOptions{
-			Container:    options.ContainerName,
-			SinceSeconds: options.SinceSeconds,
-			TailLines:    options.TailLines,
-			Follow:       options.Follow,
-		}).
-		Stream(ctx)
+func (p *Platform) ProxyFunctionLogs(ctx context.Context,
+	options interface{}) (io.ReadCloser, error) {
+	switch typedOptions := options.(type) {
+	case *platform.GetFunctionReplicaLogsStreamOptions:
+		return p.consumer.KubeClientSet.
+			CoreV1().
+			Pods(typedOptions.Namespace).
+			GetLogs(typedOptions.Name, &v1.PodLogOptions{
+				Container:    typedOptions.ContainerName,
+				SinceSeconds: typedOptions.SinceSeconds,
+				TailLines:    typedOptions.TailLines,
+				Follow:       typedOptions.Follow,
+			}).
+			Stream(ctx)
+	case *platform.ProxyFunctionLogsOptions:
+		switch typedOptions.Source {
+		case platform.ProxyLogsSourceES:
+			return p.elasticSearchClient.ProxyFunctionLogs(ctx, typedOptions)
+		case platform.ProxyLogsSourceK8s:
+			// TODO: add k8s proxy logs
+			// for now just return an error and recommend using another API
+			return nil, errors.New("K8s proxy logs not implemented, please use /logs/{replicaName} api")
```

Comment:
I thought that the idea is that the first case (using the KubeClient) will be the K8s logs source.
Is that the plan or will we keep them separate due to the different responses?

Reply:
@TomerShor I think it's better to be done in separate PR, it's still possible to support k8s logs source but with some limitations. We will need to think about translation of parameters for proxy-logs to the ones that k8s api understands.